### PR TITLE
First attempt at data persistancy

### PR DESCRIPTION
**Pull request by PaulSchweizer**
Resolves: #114 

Please take a look if this is a solution for you @Onandon11 

The flag data_persistance in Graph.evaluate can now be set. It sets the value of all plugs with a connection to None after the entire graph has been evaluated to remove the instance references.
Let me know what you think.
Another approach would be to do this during evaluation?

### CODE REVIEW

File: flowpipe/graph.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def accepts_connection(self, output_plug, input_plug):
         return True
 
     def evaluate(self, mode="linear", skip_clean=False,
-                 submission_delay=0.1, max_workers=None):
+                 submission_delay=0.1, max_workers=None, data_persistant=True):
```
Comment (neuneck) [inline flowpipe/graph.py:219]: I would suggest calling the parameter `data_persistence` to keep in line with substantives as parameter names as far as reasonably possible.

**Comment by Onandon11** [comment]
Hi Paul, Sorry this late response but due to the current situation, I didn't had the time to look at this PR. I will today! Nevertheless I hope you are doing fine..

**Comment by Onandon11** [comment]
I think this is great! I was only wondering if you print the graph all the in- and outputs are now also `None`. Maybe store the `__repr__` of the Graph before deleting all the subsequent in- and outputs?

**Comment by PaulSchweizer** [comment]
@Onandon11, I thought about storing the repr but couldn't come up with a good solution. Instead I propose to store the value of ouput plugs in the repr as well (I don't actually know why I didn't do this from the very beginning...). Check out #133 and let me know if this would resolve your concerns

File: flowpipe/graph.py (hunk 1 above)
Comment (PaulSchweizer) [reply flowpipe/graph.py]: Good point, that is done now

**Comment by PaulSchweizer** [comment]
Sorry for the long silence on this. With the merge of #133 , I think all concerns should be addressed, but please let me know if there is anything else missing here.

**Comment by Onandon11** [comment]
> Sorry for the long silence on this. With the merge of #133 , I think all concerns should be addressed, but please let me know if there is anything else missing here.

@PaulSchweizer I thinks it's fine like this :)

**Pull request merged**